Makefile

```diff
@@ -57,18 +58,88 @@ sgx: sdk sgx-env
+	cd veracruz-server && RUSTFLAGS=$(SGX_RUST_FLAG) cargo build --features sgx --features cli
+	cd veracruz-client && RUSTFLAGS=$(SGX_RUST_FLAG) cargo build --features sgx --features cli
+	# build CLIs in the SDK/test-collateral
+	$(MAKE) -C sdk/freestanding-execution-engine
+	$(MAKE) -C sdk/wasm-checker
+	$(MAKE) -C test-collateral/generate-policy
+
 nitro: sdk
 	pwd
 	RUSTFLAGS=$(NITRO_RUST_FLAG) $(MAKE) -C runtime-manager nitro
 	RUSTFLAGS=$(NITRO_RUST_FLAG) $(MAKE) -C nitro-root-enclave
 	RUSTFLAGS=$(NITRO_RUST_FLAG) $(MAKE) -C nitro-root-enclave-server
 
+nitro-cli:
+	# enclave binaries needed for veracruz-server
+	pwd
+	RUSTFLAGS=$(NITRO_RUST_FLAG) $(MAKE) -C runtime-manager nitro
+	RUSTFLAGS=$(NITRO_RUST_FLAG) $(MAKE) -C nitro-root-enclave
+	RUSTFLAGS=$(NITRO_RUST_FLAG) $(MAKE) -C nitro-root-enclave-server
+	# build CLIs in top-level crates
+	cd proxy-attestation-server && RUSTFLAGS=$(SGX_RUST_FLAG) cargo build --features nitro --features cli
+	cd veracruz-server && RUSTFLAGS=$(SGX_RUST_FLAG) cargo build --features nitro --features cli
+	cd veracruz-client && RUSTFLAGS=$(SGX_RUST_FLAG) cargo build --features nitro --features cli
+	# build CLIs in the SDK/test-collateral
+	$(MAKE) -C sdk/freestanding-execution-engine
+	$(MAKE) -C sdk/wasm-checker
+	$(MAKE) -C test-collateral/generate-policy
+
 # Compile for trustzone, note: source the rust-optee-trustzone-sdk/environment first, however assume `unset CC`.
 trustzone: sdk trustzone-env
 	$(MAKE) -C runtime-manager trustzone CC=$(AARCH64_GCC) OPTEE_DIR=$(OPTEE_DIR) OPTEE_OS_DIR=$(OPTEE_OS_DIR)
 	$(MAKE) -C trustzone-root-enclave trustzone OPTEE_DIR=$(OPTEE_DIR) OPTEE_OS_DIR=$(OPTEE_OS_DIR)
 	cd veracruz-client && RUSTFLAGS=$(SGX_RUST_FLAG) cargo build --lib --features tz
 
+trustzone-cli: trustzone-env
+	# enclave binaries needed for veracruz-server
+	$(MAKE) -C runtime-manager trustzone CC=$(AARCH64_GCC) OPTEE_DIR=$(OPTEE_DIR) OPTEE_OS_DIR=$(OPTEE_OS_DIR)
+	$(MAKE) -C trustzone-root-enclave trustzone OPTEE_DIR=$(OPTEE_DIR) OPTEE_OS_DIR=$(OPTEE_OS_DIR)
+	# build CLIs in top-level crates
+	cd proxy-attestation-server && \
+		CC_aarch64_unknown_linux_gnu=$(AARCH64_GCC) \
+		OPENSSL_INCLUDE_DIR=$(OPENSSL_INCLUDE_DIR) \
+		OPENSSL_LIB_DIR=$(OPENSSL_LIB_DIR) \
+		C_INCLUDE_PATH=$(TRUSTZONE_C_INCLUDE_PATH) \
+		cargo build --target aarch64-unknown-linux-gnu --features psa --features cli
+	cd veracruz-server && \
+		CC_aarch64_unknown_linux_gnu=$(AARCH64_GCC) \
+		OPENSSL_INCLUDE_DIR=$(OPENSSL_INCLUDE_DIR) \
+		OPENSSL_LIB_DIR=$(OPENSSL_LIB_DIR) \
+		C_INCLUDE_PATH=$(TRUSTZONE_C_INCLUDE_PATH) \
+		cargo build --target aarch64-unknown-linux-gnu --features tz --features cli
+	cd veracruz-client && RUSTFLAGS=$(SGX_RUST_FLAG) cargo build --features tz --features cli
+	# build CLIs in the SDK/test-collateral
+	$(MAKE) -C sdk/freestanding-execution-engine
+	$(MAKE) -C sdk/wasm-checker
+	$(MAKE) -C test-collateral/generate-policy
+
+%-cli-install: %-cli
```

Comment:
That's an interesting question, I haven't seen a `make install` not be phony before.

One issue that comes up is you may want to run `make sgx-cli-install` because you or something else messed with the existing installed binaries. Such a case would think there's no work to do because the dest is more recent, which is inaccurate. This could easily happen if you're switching between sgx/trustzone/nitro in the same machine.

It's worth noting `%-cli-install` doesn't actually do any compilation itself. It depends on `%-cli` which should not need to be rebuilt if the binary files are up to date.